pandas/core/generic.py

```diff
@@ -3450,6 +3450,17 @@ def to_csv(
         ...                         archive_name='out.csv')  # doctest: +SKIP
         >>> df.to_csv('out.zip', index=False,
         ...           compression=compression_opts)  # doctest: +SKIP
+
+        Create 'out.csv' in a new folder 'folder/subfolder/'
+
+        >>> import os
+        >>> os.mkdir('folder/subfolder', parents=True, exists_ok=True)
```

Comment:
os.mkdir doesn't support `parents` or `exists_ok` please try running the code before submitting a PR. 

https://docs.python.org/3/library/os.html?highlight=os%20mkdir#os.mkdir

Reply:
Oops, I copied it wrongly from [here](https://stackoverflow.com/a/11101867/14681457).

Reply:
Please check code works before committing :)